src/sys/event.rs

```diff
@@ -3,278 +3,237 @@
 
 use {Errno, Result};
 #[cfg(not(target_os = "netbsd"))]
-use libc::{timespec, time_t, c_int, c_long, uintptr_t};
+use libc::{timespec, time_t, c_int, c_long, intptr_t, uintptr_t};
 #[cfg(target_os = "netbsd")]
-use libc::{timespec, time_t, c_long, uintptr_t, size_t};
+use libc::{timespec, time_t, c_long, intptr_t, uintptr_t, size_t};
+use libc;
 use std::os::unix::io::RawFd;
 use std::ptr;
 
-pub use self::ffi::kevent as KEvent;
-
-mod ffi {
-    pub use libc::{c_int, c_void, uintptr_t, intptr_t, timespec, size_t, int64_t};
-    use super::{EventFilter, EventFlag, FilterFlag};
-
-    #[cfg(not(target_os = "netbsd"))]
-    #[derive(Clone, Copy)]
-    #[repr(C)]
-    pub struct kevent {
-        pub ident: uintptr_t,       // 8
-        pub filter: EventFilter,    // 2
-        pub flags: EventFlag,       // 2
-        pub fflags: FilterFlag,     // 4
-        pub data: intptr_t,         // 8
-        pub udata: usize             // 8
-    }
-
-    #[cfg(target_os = "netbsd")]
-    #[derive(Clone, Copy)]
-    #[repr(C)]
-    pub struct kevent {
-        pub ident: uintptr_t,
-        pub filter: EventFilter,
-        pub flags: EventFlag,
-        pub fflags: FilterFlag,
-        pub data: int64_t,
-        pub udata: intptr_t
-    }
-
-    // Bug in rustc, cannot determine that kevent is #[repr(C)]
-    #[allow(improper_ctypes)]
-    extern {
-        pub fn kqueue() -> c_int;
-
-        #[cfg(not(target_os = "netbsd"))]
-        pub fn kevent(
-            kq: c_int,
-            changelist: *const kevent,
-            nchanges: c_int,
-            eventlist: *mut kevent,
-            nevents: c_int,
-            timeout: *const timespec) -> c_int;
-
-        #[cfg(target_os = "netbsd")]
-        pub fn kevent(
-            kq: c_int,
-            changelist: *const kevent,
-            nchanges: size_t,
-            eventlist: *mut kevent,
-            nevents: size_t,
-            timeout: *const timespec) -> c_int;
-    }
+// Redefine kevent in terms of programmer-friendly enums and bitfields.
```

Comment:
@fiveop that's only because EpollEvent and EpollEventKind are both defined within Nix.  But kevent is defined within libc, and libc doesn't do enums, only constants.  That's why I redefined KEvent within nix.